pkg/distributor/distributor_test.go

```diff
@@ -96,7 +96,7 @@ func TestDistributor_Push(t *testing.T) {
 				limits.IngestionBurstSize = 20
 
 				d, _ := prepare(t, tc.numIngesters, tc.happyIngesters, 0, shardByAllLabels, limits, nil)
-				defer d.Stop()
+				defer d.StopAsync()
```

Comment:
I don't think this can introduce flaky tests. As you point out, it removes waiting, but different tests cannot interact with each other – we mock everything around them. And when used with defer, it's clear that we don't test this stopping behaviour.

Reply:
I've changed all the tests to do stop and wait on services.